cmd/sync/aws.go

```diff
@@ -61,16 +61,16 @@ func NewAWSClient(data []byte) (*AWSClient, error) {
 // GetUpstreams returns the Upstreams list
 func (client *AWSClient) GetUpstreams() []Upstream {
 	var upstreams []Upstream
-	for _, awsU := range client.config.Upstreams {
+	for i := 0; i < len(client.config.Upstreams); i++ {
 		u := Upstream{
-			Name:         awsU.Name,
-			Port:         awsU.Port,
-			Kind:         awsU.Kind,
-			ScalingGroup: awsU.AutoscalingGroup,
-			MaxConns:     &awsU.MaxConns,
-			MaxFails:     &awsU.MaxFails,
-			FailTimeout:  awsU.FailTimeout,
-			SlowStart:    awsU.SlowStart,
+			Name:         client.config.Upstreams[i].Name,
+			Port:         client.config.Upstreams[i].Port,
+			Kind:         client.config.Upstreams[i].Kind,
+			ScalingGroup: client.config.Upstreams[i].AutoscalingGroup,
+			MaxConns:     &client.config.Upstreams[i].MaxConns,
+			MaxFails:     &client.config.Upstreams[i].MaxFails,
+			FailTimeout:  getFailTimeoutOrDefault(client.config.Upstreams[i].FailTimeout),
+			SlowStart:    getSlowStartOrDefault(client.config.Upstreams[i].SlowStart),
```

Comment:
default slow start is "0s".  if the SlowStart is empty, that field won't be set the API calls when adding the server. https://github.com/nginxinc/nginx-plus-go-client/blob/master/client/nginx.go#L543 handles the comparison correctly in the update methods. is it really necessary to have the `getSlowStartOrDefault` method for it?

Reply:
Yes, because `omitempty`.

If an upstream is configured with `slow_start 10s` for instance, and they you remove the `slow_start` key from asg config and reload, the asg agent won't send anything, however, it should send the default parameter.

Please keep in mind the method you described only returns true or false if the upstreams are different, but the way the asg agent works is `empty` means `default`, that's why we need this.

Reply:
PS: This has been tested by QA and confirmed that needs to be explicitly set here when value is missing in the cfg.